sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -514,6 +517,15 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
         case _ => super.simpleString
       }
   }
+
+  def setTableNamePreprocessor(tableNamePreprocessor: (String) => String): Unit = {
+    catalog.setTableNamePreprocessor(tableNamePreprocessor)
+  }
+
+  def setHadoopFileSelector(hadoopFileSelector: Option[HadoopFileSelector]): Unit = {
+    this.hadoopFileSelector = hadoopFileSelector
+  }
```

Comment:
It just looks a little odd to me to `set` using an `Option` -- i.e. to `setHadoopFileSelector(maybeAHadoopFileSelector)` -- instead of to `set` with an actual instance and to explicitly clear instead of to set to None.  I guess what I am saying is that it makes sense for the underlying `this.hadoopFileSelector` to be an Option (maybe there, maybe not), but that when setting or removing the `hadoopFileSelector` the caller of the method(s) would naturally have a concrete idea of what should be done and wrapping that concreteness in a `maybe` doesn't make obvious sense or improve the readability at the callsite of the set/unset.